src/app/util/af-event.cpp

```diff
@@ -28,8 +28,6 @@
 
 #include <app-common/zap-generated/callback.h>
 
```

Comment:
I am not sure we can do this:

zzz_generated/tv-casting-app/zap-generated/af-gen-event.h for example defines EMBER_AF_GENERATED_EVENT_CODE so things may break.

I am actually not sure if this is actually used for tv, but overall the comment is we need to identify usages and determine if any of them need it. if nothing needs it, we could just delete af-event alltogether.

```sh
> rg EMBER_AF_GENERATED_EVENT_CODE --files-with-matches | grep -v third_party
zzz_generated/all-clusters-minimal-app/zap-generated/af-gen-event.h
zzz_generated/tv-casting-app/zap-generated/af-gen-event.h
zzz_generated/all-clusters-app/zap-generated/af-gen-event.h
zzz_generated/controller-clusters/zap-generated/af-gen-event.h
src/app/util/af-event.cpp
```

```sh
> rg EMBER_AF_GENERATED_EVENTS --files-with-matches | grep -v third_party
zzz_generated/tv-casting-app/zap-generated/af-gen-event.h
zzz_generated/all-clusters-minimal-app/zap-generated/af-gen-event.h
zzz_generated/all-clusters-app/zap-generated/af-gen-event.h
src/app/util/af-event.cpp
zzz_generated/controller-clusters/zap-generated/af-gen-event.h
```

Reply:
From what I can see, what gets generated includes:

```
emberAfPluginDoorLockServerLockoutEventHandler
emberAfBarrierControlClusterServerTickCallbackControl1
```

and tv also has:

```
emberAfPluginColorControlServerTempTransitionEventHandler
emberAfPluginColorControlServerXyTransitionEventHandler
```

we probably have to see if these are needed (why does a TV need temperature transitions and xy? @chrisdecenzo 

and beyond that ... if the clusters implementations rely on af-events we have to transition them to System timers probably (my understanding of events is that they are generally timers)

Reply:
The original implementation of some clusters was relying on af-gen-event to generate ticks. Level cluster was one cluster that we refactored to remove dependency of af-gen-event

Reply:
[TV casting app's af-gen-event](https://github.com/project-chip/connectedhomeip/blob/master/zzz_generated/tv-casting-app/zap-generated/af-gen-event.h) has references what seem unrelated clusters
- BarrierControl
- ColorControl
- DoorLock

Looks more like some code left over from a previous sample that was used as a starting point.

All-clusters also have some references that look legacy.

I would only be concerned about some real dependency on the barrier control cluster, but the lock app doesn't use af-gen-event.h

Reply:
Btw Barrier Control is not in Matter and will be likely being deprecated in the future; just saying

Reply:
In that case, my reasoning seems to be that we can indeed remove it:

   - Barrier control should not be there
   - Door lock seems invalid because we use 'door lock app' to test and only all clusters has it 
   - ColorControl seems invalid in a TV app